### PR TITLE
chore(dogfood): update dogfood template to use artifactory

### PR DESCRIPTION
**Pull request by matifali**
This pull request updates the dogfood template to configure `npm`, `pip`, `go` and `docker` to fetch packages from the Artifactory. This also installs the JFrog extension in code-server.

### CODE REVIEW

File: dogfood/main.tf [hunk 1]
```diff
@@ -250,22 +283,16 @@ resource "docker_volume" "home_volume" {
...
-  pull_triggers = [
-    data.docker_registry_image.dogfood.sha256_digest,
-    sha1(join("", [for f in fileset(path.module, "files/*") : filesha1(f)])),
-    filesha1("Dockerfile"),
-  ]
-  keep_locally = true
+  name          = "${local.jfrog_host}/docker/${local.registry_name}:latest"
+  pull_triggers = [null_resource.update_trigger.id]
+  keep_locally  = true
 }
```
Comment (matifali) [inline dogfood/main.tf:296]: This was not working nicely with a private docker registry, so I had to resort to this hack,
I filed an issue with the docker provider.
https://github.com/kreuzwerker/terraform-provider-docker/issues/593

Comment (matifali) [reply dogfood/main.tf:296]: This is also not working, and terraform is not pulling the latest image even though Artifactory has it.
There are open issues regarding this: https://github.com/kreuzwerker/terraform-provider-docker/issues/172

Comment (matifali) [reply dogfood/main.tf:296]: We can skip pulling from the satisfactory in Terraform or use a new tag on each build. Probably short git sha.

**Review by Emyrk — APPROVED** [review]

File: dogfood/main.tf (hunk 1 above)
Comment (matifali) [reply dogfood/main.tf]: After talking to @Emyrk, we decided not to use Artifactory to fetch pull docker image.

**Pull request merged**